src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignedCms.cs

```diff
@@ -248,7 +248,16 @@ public void ComputeSignature(CmsSigner signer, bool silent)
             {
                 throw new CryptographicException(SR.Cryptography_Cms_Sign_Empty_Content);
             }
-            
+
+            if (signer.SignerIdentifierType != SubjectIdentifierType.NoSignature &&
+                signer.Certificate == null)
+            {
+                if (silent)
+                    throw new InvalidOperationException(SR.Cryptography_Cms_RecipientCertificateNotFound);
```

Comment:
recipient? I think this should be signer

Reply:
Hmm, I took the name and text from NetFX.

Reply:
yes, it's not correct on netfx but I think corefx can fix it as long as exception type is the same

Reply:
Agreed, I will do that.